tests/e2e/flowtest/test_cephfilesystem_creation.py

```diff
@@ -0,0 +1,48 @@
+import pytest
+import logging
+
+from ocs_ci.ocs import constants, defaults
+from ocs_ci.helpers.helpers import create_resource, validate_cephfilesystem
+from ocs_ci.ocs.exceptions import ResourceAlreadyExists
+from ocs_ci.utility import templating
+
+
+logger = logging.getLogger(__name__)
+cephFS_data = templating.load_yaml(constants.CEPHFILESYSTEM_YAML)
+cephFS_data["metadata"]["name"] = "test-ceph-fs"
+cephFS_data["metadata"]["namespace"] = defaults.ROOK_CLUSTER_NAMESPACE
+
+
+class TestCephFileSystemCreation:
+    """
+    Testing Creation of a filesystem and checking its existence.
+    Also checking if the same filesystem can be created more than once or not.
+    """
+
+    @pytest.fixture()
+    def cephFileSystem(self):
+        logging.info("Creating CephFileSystem in the namespace")
+        cephFS_obj = create_resource(**cephFS_data)
+        cephFS_obj.reload()
+        assert validate_cephfilesystem(
+            cephFS_obj.name
+        ), f"File system {cephFS_obj.name} does not exist"
+        logging.info("CephFile System Created. : testCephFS")
+        yield cephFS_obj
+        logging.info(
+            "Teardown of the test, deleting the test Ceph File System testCephFS"
+        )
+        cephFS_obj.delete()
```

Comment:
IIUC, with this logic cephFS_obj gets deleted before the actual test is done.

Reply:
This doesn't delete the cephFS_obj because of the yield statement before it.
Its recommended by pytest to use yield as a teardown method if needed.
https://docs.pytest.org/en/6.2.x/fixture.html#yield-fixtures-recommended